cmd/uplink/cmd/cp.go

```diff
@@ -37,33 +36,19 @@ func init() {
 	progress = cpCmd.Flags().Bool("progress", true, "if true, show progress")
 }
 
-func cleanAbsPath(p string) string {
-	prefix := strings.HasSuffix(p, "/")
-	p = path.Join("/", p)
-	if !strings.HasSuffix(p, "/") && prefix {
-		p += "/"
-	}
-	return p
-}
-
 // upload uploads args[0] from local machine to s3 compatible object args[1]
-func upload(ctx context.Context, bs buckets.Store, srcFile string, destObj *url.URL) error {
-	var err error
-	if destObj.Scheme == "" {
-		return fmt.Errorf("Invalid destination")
-	}
-
-	destObj.Path = cleanAbsPath(destObj.Path)
+func upload(ctx context.Context, bs buckets.Store, srcFile fpath.FPath, destObj fpath.FPath) error {
 	// if object name not specified, default to filename
-	if strings.HasSuffix(destObj.Path, "/") {
-		destObj.Path = path.Join(destObj.Path, path.Base(srcFile))
+	if strings.HasSuffix(destObj.Path(), "/") || destObj.Path() == "" {
+		destObj = destObj.Join(srcFile.Base())
 	}
 
 	var f *os.File
-	if srcFile == "-" {
+	var err error
+	if srcFile.Base() == "-" {
```

Comment:
Should we add a special flag for `IsPipe`?

Reply:
I thought about this, but I didn't want to include it in the FPath type as it seems something very specific to the `cp` command. And it works fine as it right now.